group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -2227,81 +2611,95 @@ public CoordinatorResult<Void, Record> classicGroupJoin(
         RequestContext context,
         JoinGroupRequestData request,
         CompletableFuture<JoinGroupResponseData> responseFuture
+    ) {
+        throwIfClassicGroupSessionTimeoutInvalid(request.sessionTimeoutMs());
```

Comment:
Seeing this, I wonder if we should move all the static request validations to the group coordinator service. I does not make sense to occupy the coordinator threads for this. What do you think?